### PR TITLE
Hijack api/* to provide a better error message in case of 404

### PR DESCRIPTION
**Pull request by amitaibu**
#159

### CODE REVIEW

File: restful.module [hunk 1]
```diff
@@ -773,6 +777,40 @@ function _restful_get_json_from_menu_status(&$var) {
...
+ * Hijack api/* to be under RESTful. We make sure that any call to api/* pages
+ * that isn't valid, will still return with a well formatted error, instead of
+ * a 404 HTML page.
+ */
+function restful_page_delivery_callback_alter(&$callback) {
+  if (!variable_get('restful_hijack_api_pages', TRUE)) {
+    return;
+  }
+
+  if (strpos($_GET['q'], 'api/') !== 0) {
```
Comment (e0ipso) [inline restful.module]: We should use the contents of `'restful_hook_menu_base_path'` variable. I see that `'api/login'`, `'api/file-upload'` and `'api/session/token'` will also need fixing in this respect.

File: restful.module [hunk 2]
```diff
@@ -773,6 +777,40 @@ function _restful_get_json_from_menu_status(&$var) {
...
+
+/**
+ * Implements hook_page_delivery_callback_alter().
+ *
+ * Hijack api/* to be under RESTful. We make sure that any call to api/* pages
+ * that isn't valid, will still return with a well formatted error, instead of
+ * a 404 HTML page.
+ */
+function restful_page_delivery_callback_alter(&$callback) {
+  if (!variable_get('restful_hijack_api_pages', TRUE)) {
```
Comment (e0ipso) [inline restful.module:800]: We should document this variable and delete it on uninstall.

**Comment by e0ipso** [comment]
Maybe also worth adding a test.

**Comment by amitaibu** [comment]
Fixed comments and added tests - which are passing.

**Comment by e0ipso** [comment]
Merged via 25ad31e897846a42114cc346d51be28b52cdad09.

Thanks!

**Pull request closed without merging**